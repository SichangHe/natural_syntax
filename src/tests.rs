use super::*;

use std::time::Instant;

use insta::assert_debug_snapshot;
use tracing::{info, Level};
use tracing_subscriber::EnvFilter;

#[test]
fn paragraph_pos() {
    init_tracing();
    let start = Instant::now();
    let model = POSModel::try_default().unwrap();
    info!("Took {}ms to load the model.", start.elapsed().as_millis());
    let input = "Extracts Part of Speech tags (Noun, Verb, Adjectiveâ€¦) from text. A lightweight pretrained model using MobileBERT is available for English.";
    let start = Instant::now();
    let output = model.predict(input);
    info!("Took {}ms to predict.", start.elapsed().as_millis());
    let mut parsed = output.collect::<Result<Vec<_>, _>>().unwrap();
    round_scores(&mut parsed);
    assert_debug_snapshot!(parsed);
}

#[test]
fn markdown_pos() {
    init_tracing();
    let start = Instant::now();
    let model = POSModel::try_default().unwrap();
    info!("Took {}ms to load the model.", start.elapsed().as_millis());
    let input = "The easiest way of getting started is the [rustler Elixir library](https://hex.pm/packages/rustler).

- Add the [rustler Elixir library](https://hex.pm/packages/rustler) as a
  dependency of your project.
- Run `mix rustler.new` to generate a new NIF in your project. Follow the
  instructions.
- If you are already using [`serde`](https://serde.rs) and/or have been using
  `serde_rustler` before, please enable the `serde` feature in your NIF crate's
  `Cargo.toml` on the `rustler` dependency.
";
    let start = Instant::now();
    let output = model.predict(input);
    info!("Took {}ms to predict.", start.elapsed().as_millis());
    let mut parsed = output.collect::<Result<Vec<_>, _>>().unwrap();
    round_scores(&mut parsed);
    assert_debug_snapshot!(parsed);
}

const PRECISION: f64 = 1e-3;

fn round_scores(predictions: &mut [POSToken]) {
    predictions
        .iter_mut()
        .for_each(|token| token.score = (token.score / PRECISION).round() * PRECISION)
}

fn init_tracing() {
    _ = tracing_subscriber::fmt()
        .with_env_filter(
            EnvFilter::builder()
                .with_default_directive(Level::INFO.into())
                .from_env_lossy(),
        )
        .try_init();
}
