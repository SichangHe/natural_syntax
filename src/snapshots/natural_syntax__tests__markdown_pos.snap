---
source: src/tests.rs
expression: parsed
---
[
    POSToken {
        word: "The",
        score: 0.9998831152915955,
        tag: DT,
        label_index: 3,
        index: 1,
        word_index: 1,
        offset_begin: 0,
        offset_end: 3,
    },
    POSToken {
        word: "easiest",
        score: 0.8221415281295776,
        tag: JJS,
        label_index: 9,
        index: 2,
        word_index: 2,
        offset_begin: 4,
        offset_end: 11,
    },
    POSToken {
        word: "way",
        score: 0.9977756142616272,
        tag: NN,
        label_index: 11,
        index: 3,
        word_index: 3,
        offset_begin: 12,
        offset_end: 15,
    },
    POSToken {
        word: "of",
        score: 0.9999945163726807,
        tag: IN,
        label_index: 6,
        index: 4,
        word_index: 4,
        offset_begin: 16,
        offset_end: 18,
    },
    POSToken {
        word: "getting",
        score: 0.9960637092590332,
        tag: VBG,
        label_index: 27,
        index: 5,
        word_index: 5,
        offset_begin: 19,
        offset_end: 26,
    },
    POSToken {
        word: "started",
        score: 0.8323637843132019,
        tag: VBN,
        label_index: 28,
        index: 6,
        word_index: 6,
        offset_begin: 27,
        offset_end: 34,
    },
    POSToken {
        word: "is",
        score: 0.9998722076416016,
        tag: VBZ,
        label_index: 30,
        index: 7,
        word_index: 7,
        offset_begin: 35,
        offset_end: 37,
    },
    POSToken {
        word: "the",
        score: 0.9999052882194519,
        tag: DT,
        label_index: 3,
        index: 8,
        word_index: 8,
        offset_begin: 38,
        offset_end: 41,
    },
    POSToken {
        word: "[",
        score: 0.9384734630584717,
        tag: NNP,
        label_index: 12,
        index: 9,
        word_index: 9,
        offset_begin: 42,
        offset_end: 43,
    },
    POSToken {
        word: "rustler",
        score: 0.36435735180492657,
        tag: NNP,
        label_index: 12,
        index: 10,
        word_index: 10,
        offset_begin: 43,
        offset_end: 50,
    },
    POSToken {
        word: "Elixir",
        score: 0.1591313641145719,
        tag: NNP,
        label_index: 12,
        index: 12,
        word_index: 11,
        offset_begin: 51,
        offset_end: 57,
    },
    POSToken {
        word: "library",
        score: 0.6908725500106812,
        tag: NN,
        label_index: 11,
        index: 15,
        word_index: 12,
        offset_begin: 58,
        offset_end: 65,
    },
    POSToken {
        word: "]",
        score: 0.8425936698913574,
        tag: NNP,
        label_index: 12,
        index: 16,
        word_index: 13,
        offset_begin: 65,
        offset_end: 66,
    },
    POSToken {
        word: "(",
        score: 0.6715282797813416,
        tag: IN,
        label_index: 6,
        index: 17,
        word_index: 14,
        offset_begin: 66,
        offset_end: 67,
    },
    POSToken {
        word: "https",
        score: 0.4684299826622009,
        tag: NNS,
        label_index: 14,
        index: 18,
        word_index: 15,
        offset_begin: 67,
        offset_end: 72,
    },
    POSToken {
        word: ":",
        score: 0.539379358291626,
        tag: NNP,
        label_index: 12,
        index: 19,
        word_index: 16,
        offset_begin: 72,
        offset_end: 73,
    },
    POSToken {
        word: "/",
        score: 0.8090564012527466,
        tag: NNP,
        label_index: 12,
        index: 20,
        word_index: 17,
        offset_begin: 73,
        offset_end: 74,
    },
    POSToken {
        word: "/",
        score: 0.6449680328369141,
        tag: NNP,
        label_index: 12,
        index: 21,
        word_index: 18,
        offset_begin: 74,
        offset_end: 75,
    },
    POSToken {
        word: "hex",
        score: 0.8052503596231588,
        tag: NNP,
        label_index: 12,
        index: 22,
        word_index: 19,
        offset_begin: 75,
        offset_end: 78,
    },
    POSToken {
        word: ".",
        score: 0.8646934032440186,
        tag: NNP,
        label_index: 12,
        index: 24,
        word_index: 20,
        offset_begin: 78,
        offset_end: 79,
    },
    POSToken {
        word: "pm",
        score: 0.9731399416923523,
        tag: NNP,
        label_index: 12,
        index: 25,
        word_index: 21,
        offset_begin: 79,
        offset_end: 81,
    },
    POSToken {
        word: "/",
        score: 0.5624796748161316,
        tag: NN,
        label_index: 11,
        index: 26,
        word_index: 22,
        offset_begin: 81,
        offset_end: 82,
    },
    POSToken {
        word: "packages",
        score: 0.6462132930755615,
        tag: NNS,
        label_index: 14,
        index: 27,
        word_index: 23,
        offset_begin: 82,
        offset_end: 90,
    },
    POSToken {
        word: "/",
        score: 0.8011429905891418,
        tag: NN,
        label_index: 11,
        index: 28,
        word_index: 24,
        offset_begin: 90,
        offset_end: 91,
    },
    POSToken {
        word: "rustler",
        score: 0.2567128258759297,
        tag: NNP,
        label_index: 12,
        index: 29,
        word_index: 25,
        offset_begin: 91,
        offset_end: 98,
    },
    POSToken {
        word: ")",
        score: 0.2593020796775818,
        tag: PRP,
        label_index: 17,
        index: 31,
        word_index: 26,
        offset_begin: 98,
        offset_end: 99,
    },
    POSToken {
        word: ".",
        score: 0.29849743843078613,
        tag: NNP,
        label_index: 12,
        index: 32,
        word_index: 27,
        offset_begin: 99,
        offset_end: 100,
    },
    POSToken {
        word: "-",
        score: 0.4431455731391907,
        tag: JJ,
        label_index: 7,
        index: 33,
        word_index: 28,
        offset_begin: 102,
        offset_end: 103,
    },
    POSToken {
        word: "Add",
        score: 0.9256914854049683,
        tag: VB,
        label_index: 25,
        index: 34,
        word_index: 29,
        offset_begin: 104,
        offset_end: 107,
    },
    POSToken {
        word: "the",
        score: 0.9998955130577087,
        tag: DT,
        label_index: 3,
        index: 35,
        word_index: 30,
        offset_begin: 108,
        offset_end: 111,
    },
    POSToken {
        word: "[",
        score: 0.9037013053894043,
        tag: NNP,
        label_index: 12,
        index: 36,
        word_index: 31,
        offset_begin: 112,
        offset_end: 113,
    },
    POSToken {
        word: "rustler",
        score: 0.2997952459870987,
        tag: NNP,
        label_index: 12,
        index: 37,
        word_index: 32,
        offset_begin: 113,
        offset_end: 120,
    },
    POSToken {
        word: "Elixir",
        score: 0.21288051803588076,
        tag: NNP,
        label_index: 12,
        index: 39,
        word_index: 33,
        offset_begin: 121,
        offset_end: 127,
    },
    POSToken {
        word: "library",
        score: 0.6596149206161499,
        tag: NN,
        label_index: 11,
        index: 42,
        word_index: 34,
        offset_begin: 128,
        offset_end: 135,
    },
    POSToken {
        word: "]",
        score: 0.8854286074638367,
        tag: NNP,
        label_index: 12,
        index: 43,
        word_index: 35,
        offset_begin: 135,
        offset_end: 136,
    },
    POSToken {
        word: "(",
        score: 0.6001749038696289,
        tag: IN,
        label_index: 6,
        index: 44,
        word_index: 36,
        offset_begin: 136,
        offset_end: 137,
    },
    POSToken {
        word: "https",
        score: 0.4767860770225525,
        tag: NNS,
        label_index: 14,
        index: 45,
        word_index: 37,
        offset_begin: 137,
        offset_end: 142,
    },
    POSToken {
        word: ":",
        score: 0.4551432132720947,
        tag: NNP,
        label_index: 12,
        index: 46,
        word_index: 38,
        offset_begin: 142,
        offset_end: 143,
    },
    POSToken {
        word: "/",
        score: 0.7439953088760376,
        tag: NNP,
        label_index: 12,
        index: 47,
        word_index: 39,
        offset_begin: 143,
        offset_end: 144,
    },
    POSToken {
        word: "/",
        score: 0.5564883351325989,
        tag: NNP,
        label_index: 12,
        index: 48,
        word_index: 40,
        offset_begin: 144,
        offset_end: 145,
    },
    POSToken {
        word: "hex",
        score: 0.8010911102495726,
        tag: NNP,
        label_index: 12,
        index: 49,
        word_index: 41,
        offset_begin: 145,
        offset_end: 148,
    },
    POSToken {
        word: ".",
        score: 0.8285888433456421,
        tag: NNP,
        label_index: 12,
        index: 51,
        word_index: 42,
        offset_begin: 148,
        offset_end: 149,
    },
    POSToken {
        word: "pm",
        score: 0.9634965658187866,
        tag: NNP,
        label_index: 12,
        index: 52,
        word_index: 43,
        offset_begin: 149,
        offset_end: 151,
    },
    POSToken {
        word: "/",
        score: 0.6519892811775208,
        tag: NN,
        label_index: 11,
        index: 53,
        word_index: 44,
        offset_begin: 151,
        offset_end: 152,
    },
    POSToken {
        word: "packages",
        score: 0.6053404808044434,
        tag: NNS,
        label_index: 14,
        index: 54,
        word_index: 45,
        offset_begin: 152,
        offset_end: 160,
    },
    POSToken {
        word: "/",
        score: 0.8548048734664917,
        tag: NN,
        label_index: 11,
        index: 55,
        word_index: 46,
        offset_begin: 160,
        offset_end: 161,
    },
    POSToken {
        word: "rustler",
        score: 0.29196474704283304,
        tag: NNP,
        label_index: 12,
        index: 56,
        word_index: 47,
        offset_begin: 161,
        offset_end: 168,
    },
    POSToken {
        word: ")",
        score: 0.3050229549407959,
        tag: NNP,
        label_index: 12,
        index: 58,
        word_index: 48,
        offset_begin: 168,
        offset_end: 169,
    },
    POSToken {
        word: "as",
        score: 0.9935842156410217,
        tag: IN,
        label_index: 6,
        index: 59,
        word_index: 49,
        offset_begin: 170,
        offset_end: 172,
    },
    POSToken {
        word: "a",
        score: 0.999955415725708,
        tag: DT,
        label_index: 3,
        index: 60,
        word_index: 50,
        offset_begin: 173,
        offset_end: 174,
    },
    POSToken {
        word: "dependency",
        score: 0.9978703260421753,
        tag: NN,
        label_index: 11,
        index: 61,
        word_index: 51,
        offset_begin: 177,
        offset_end: 187,
    },
    POSToken {
        word: "of",
        score: 0.9999860525131226,
        tag: IN,
        label_index: 6,
        index: 62,
        word_index: 52,
        offset_begin: 188,
        offset_end: 190,
    },
    POSToken {
        word: "your",
        score: 0.7856572270393372,
        tag: PRP,
        label_index: 17,
        index: 63,
        word_index: 53,
        offset_begin: 191,
        offset_end: 195,
    },
    POSToken {
        word: "project",
        score: 0.9944018125534058,
        tag: NN,
        label_index: 11,
        index: 64,
        word_index: 54,
        offset_begin: 196,
        offset_end: 203,
    },
    POSToken {
        word: ".",
        score: 0.30855053663253784,
        tag: NNS,
        label_index: 14,
        index: 65,
        word_index: 55,
        offset_begin: 203,
        offset_end: 204,
    },
    POSToken {
        word: "-",
        score: 0.23409134149551392,
        tag: JJ,
        label_index: 7,
        index: 66,
        word_index: 56,
        offset_begin: 205,
        offset_end: 206,
    },
    POSToken {
        word: "Run",
        score: 0.9588191509246826,
        tag: VB,
        label_index: 25,
        index: 67,
        word_index: 57,
        offset_begin: 207,
        offset_end: 210,
    },
    POSToken {
        word: "`",
        score: 0.7637679576873779,
        tag: NN,
        label_index: 11,
        index: 68,
        word_index: 58,
        offset_begin: 211,
        offset_end: 212,
    },
    POSToken {
        word: "mix",
        score: 0.7935993671417236,
        tag: VB,
        label_index: 25,
        index: 69,
        word_index: 59,
        offset_begin: 212,
        offset_end: 215,
    },
    POSToken {
        word: "rustler",
        score: 0.24895743390845837,
        tag: NNP,
        label_index: 12,
        index: 70,
        word_index: 60,
        offset_begin: 216,
        offset_end: 223,
    },
    POSToken {
        word: ".",
        score: 0.2474227249622345,
        tag: NNP,
        label_index: 12,
        index: 72,
        word_index: 61,
        offset_begin: 223,
        offset_end: 224,
    },
    POSToken {
        word: "new",
        score: 0.524986207485199,
        tag: JJ,
        label_index: 7,
        index: 73,
        word_index: 62,
        offset_begin: 224,
        offset_end: 227,
    },
    POSToken {
        word: "`",
        score: 0.5661125183105469,
        tag: NNS,
        label_index: 14,
        index: 74,
        word_index: 63,
        offset_begin: 227,
        offset_end: 228,
    },
    POSToken {
        word: "to",
        score: 0.9999556541442871,
        tag: TO,
        label_index: 23,
        index: 75,
        word_index: 64,
        offset_begin: 229,
        offset_end: 231,
    },
    POSToken {
        word: "generate",
        score: 0.9902148842811584,
        tag: VB,
        label_index: 25,
        index: 76,
        word_index: 65,
        offset_begin: 232,
        offset_end: 240,
    },
    POSToken {
        word: "a",
        score: 0.9999816417694092,
        tag: DT,
        label_index: 3,
        index: 77,
        word_index: 66,
        offset_begin: 241,
        offset_end: 242,
    },
    POSToken {
        word: "new",
        score: 0.9973697662353516,
        tag: JJ,
        label_index: 7,
        index: 78,
        word_index: 67,
        offset_begin: 243,
        offset_end: 246,
    },
    POSToken {
        word: "NIF",
        score: 0.4136504369867602,
        tag: NN,
        label_index: 11,
        index: 79,
        word_index: 68,
        offset_begin: 247,
        offset_end: 250,
    },
    POSToken {
        word: "in",
        score: 0.9997164011001587,
        tag: IN,
        label_index: 6,
        index: 81,
        word_index: 69,
        offset_begin: 251,
        offset_end: 253,
    },
    POSToken {
        word: "your",
        score: 0.6077812910079956,
        tag: PRP,
        label_index: 17,
        index: 82,
        word_index: 70,
        offset_begin: 254,
        offset_end: 258,
    },
    POSToken {
        word: "project",
        score: 0.9970571398735046,
        tag: NN,
        label_index: 11,
        index: 83,
        word_index: 71,
        offset_begin: 259,
        offset_end: 266,
    },
    POSToken {
        word: ".",
        score: 0.2475094348192215,
        tag: TO,
        label_index: 23,
        index: 84,
        word_index: 72,
        offset_begin: 266,
        offset_end: 267,
    },
    POSToken {
        word: "Follow",
        score: 0.9774355292320251,
        tag: VB,
        label_index: 25,
        index: 85,
        word_index: 73,
        offset_begin: 268,
        offset_end: 274,
    },
    POSToken {
        word: "the",
        score: 0.9999900460243225,
        tag: DT,
        label_index: 3,
        index: 86,
        word_index: 74,
        offset_begin: 275,
        offset_end: 278,
    },
    POSToken {
        word: "instructions",
        score: 0.9945311546325684,
        tag: NNS,
        label_index: 14,
        index: 87,
        word_index: 75,
        offset_begin: 281,
        offset_end: 293,
    },
    POSToken {
        word: ".",
        score: 0.43976157903671265,
        tag: NNS,
        label_index: 14,
        index: 88,
        word_index: 76,
        offset_begin: 293,
        offset_end: 294,
    },
    POSToken {
        word: "-",
        score: 0.6518104672431946,
        tag: JJ,
        label_index: 7,
        index: 89,
        word_index: 77,
        offset_begin: 295,
        offset_end: 296,
    },
    POSToken {
        word: "If",
        score: 0.9914919137954712,
        tag: IN,
        label_index: 6,
        index: 90,
        word_index: 78,
        offset_begin: 297,
        offset_end: 299,
    },
    POSToken {
        word: "you",
        score: 0.9744523763656616,
        tag: PRP,
        label_index: 17,
        index: 91,
        word_index: 79,
        offset_begin: 300,
        offset_end: 303,
    },
    POSToken {
        word: "are",
        score: 0.9339438676834106,
        tag: VBP,
        label_index: 29,
        index: 92,
        word_index: 80,
        offset_begin: 304,
        offset_end: 307,
    },
    POSToken {
        word: "already",
        score: 0.9963840246200562,
        tag: RB,
        label_index: 18,
        index: 93,
        word_index: 81,
        offset_begin: 308,
        offset_end: 315,
    },
    POSToken {
        word: "using",
        score: 0.9983450174331665,
        tag: VBG,
        label_index: 27,
        index: 94,
        word_index: 82,
        offset_begin: 316,
        offset_end: 321,
    },
    POSToken {
        word: "[",
        score: 0.8916382789611816,
        tag: NNP,
        label_index: 12,
        index: 95,
        word_index: 83,
        offset_begin: 322,
        offset_end: 323,
    },
    POSToken {
        word: "`",
        score: 0.5970437526702881,
        tag: NNP,
        label_index: 12,
        index: 96,
        word_index: 84,
        offset_begin: 323,
        offset_end: 324,
    },
    POSToken {
        word: "serde",
        score: 0.5468462943803338,
        tag: NNP,
        label_index: 12,
        index: 97,
        word_index: 85,
        offset_begin: 324,
        offset_end: 329,
    },
    POSToken {
        word: "`",
        score: 0.7715820670127869,
        tag: NNP,
        label_index: 12,
        index: 99,
        word_index: 86,
        offset_begin: 329,
        offset_end: 330,
    },
    POSToken {
        word: "]",
        score: 0.7333847284317017,
        tag: NNP,
        label_index: 12,
        index: 100,
        word_index: 87,
        offset_begin: 330,
        offset_end: 331,
    },
    POSToken {
        word: "(",
        score: 0.49356919527053833,
        tag: IN,
        label_index: 6,
        index: 101,
        word_index: 88,
        offset_begin: 331,
        offset_end: 332,
    },
    POSToken {
        word: "https",
        score: 0.4803343415260315,
        tag: NNS,
        label_index: 14,
        index: 102,
        word_index: 89,
        offset_begin: 332,
        offset_end: 337,
    },
    POSToken {
        word: ":",
        score: 0.42196065187454224,
        tag: NNP,
        label_index: 12,
        index: 103,
        word_index: 90,
        offset_begin: 337,
        offset_end: 338,
    },
    POSToken {
        word: "/",
        score: 0.6600311398506165,
        tag: NNP,
        label_index: 12,
        index: 104,
        word_index: 91,
        offset_begin: 338,
        offset_end: 339,
    },
    POSToken {
        word: "/",
        score: 0.4615519940853119,
        tag: NNP,
        label_index: 12,
        index: 105,
        word_index: 92,
        offset_begin: 339,
        offset_end: 340,
    },
    POSToken {
        word: "serde",
        score: 0.7899681361461575,
        tag: NNP,
        label_index: 12,
        index: 106,
        word_index: 93,
        offset_begin: 340,
        offset_end: 345,
    },
    POSToken {
        word: ".",
        score: 0.8450068831443787,
        tag: NNP,
        label_index: 12,
        index: 108,
        word_index: 94,
        offset_begin: 345,
        offset_end: 346,
    },
    POSToken {
        word: "rs",
        score: 0.6719708442687988,
        tag: NNP,
        label_index: 12,
        index: 109,
        word_index: 95,
        offset_begin: 346,
        offset_end: 348,
    },
    POSToken {
        word: ")",
        score: 0.2542460560798645,
        tag: NN,
        label_index: 11,
        index: 110,
        word_index: 96,
        offset_begin: 348,
        offset_end: 349,
    },
    POSToken {
        word: "and",
        score: 0.999634861946106,
        tag: CC,
        label_index: 1,
        index: 111,
        word_index: 97,
        offset_begin: 350,
        offset_end: 353,
    },
    POSToken {
        word: "/",
        score: 0.9009062051773071,
        tag: NN,
        label_index: 11,
        index: 112,
        word_index: 98,
        offset_begin: 353,
        offset_end: 354,
    },
    POSToken {
        word: "or",
        score: 0.9848076105117798,
        tag: CC,
        label_index: 1,
        index: 113,
        word_index: 99,
        offset_begin: 354,
        offset_end: 356,
    },
    POSToken {
        word: "have",
        score: 0.555888295173645,
        tag: VBP,
        label_index: 29,
        index: 114,
        word_index: 100,
        offset_begin: 357,
        offset_end: 361,
    },
    POSToken {
        word: "been",
        score: 0.9985567927360535,
        tag: VBN,
        label_index: 28,
        index: 115,
        word_index: 101,
        offset_begin: 362,
        offset_end: 366,
    },
    POSToken {
        word: "using",
        score: 0.9925937652587891,
        tag: VBG,
        label_index: 27,
        index: 116,
        word_index: 102,
        offset_begin: 367,
        offset_end: 372,
    },
    POSToken {
        word: "`",
        score: 0.3011713922023773,
        tag: NN,
        label_index: 11,
        index: 117,
        word_index: 103,
        offset_begin: 375,
        offset_end: 376,
    },
    POSToken {
        word: "serde",
        score: 0.7325014068106839,
        tag: NNP,
        label_index: 12,
        index: 118,
        word_index: 104,
        offset_begin: 376,
        offset_end: 381,
    },
    POSToken {
        word: "_",
        score: 0.42046573758125305,
        tag: NN,
        label_index: 11,
        index: 120,
        word_index: 105,
        offset_begin: 381,
        offset_end: 382,
    },
    POSToken {
        word: "rustler",
        score: 0.36768210407343815,
        tag: NNP,
        label_index: 12,
        index: 121,
        word_index: 106,
        offset_begin: 382,
        offset_end: 389,
    },
    POSToken {
        word: "`",
        score: 0.7340719103813171,
        tag: NNS,
        label_index: 14,
        index: 123,
        word_index: 107,
        offset_begin: 389,
        offset_end: 390,
    },
    POSToken {
        word: "before",
        score: 0.9265612959861755,
        tag: IN,
        label_index: 6,
        index: 124,
        word_index: 108,
        offset_begin: 391,
        offset_end: 397,
    },
    POSToken {
        word: ",",
        score: 0.1866411715745926,
        tag: DT,
        label_index: 3,
        index: 125,
        word_index: 109,
        offset_begin: 397,
        offset_end: 398,
    },
    POSToken {
        word: "please",
        score: 0.6984201073646545,
        tag: VB,
        label_index: 25,
        index: 126,
        word_index: 110,
        offset_begin: 399,
        offset_end: 405,
    },
    POSToken {
        word: "enable",
        score: 0.9540420174598694,
        tag: VB,
        label_index: 25,
        index: 127,
        word_index: 111,
        offset_begin: 406,
        offset_end: 412,
    },
    POSToken {
        word: "the",
        score: 0.9999853372573853,
        tag: DT,
        label_index: 3,
        index: 128,
        word_index: 112,
        offset_begin: 413,
        offset_end: 416,
    },
    POSToken {
        word: "`",
        score: 0.46623894572257996,
        tag: NN,
        label_index: 11,
        index: 129,
        word_index: 113,
        offset_begin: 417,
        offset_end: 418,
    },
    POSToken {
        word: "serde",
        score: 0.5342979453321632,
        tag: NNP,
        label_index: 12,
        index: 130,
        word_index: 114,
        offset_begin: 418,
        offset_end: 423,
    },
    POSToken {
        word: "`",
        score: 0.5512051582336426,
        tag: NNS,
        label_index: 14,
        index: 132,
        word_index: 115,
        offset_begin: 423,
        offset_end: 424,
    },
    POSToken {
        word: "feature",
        score: 0.981834352016449,
        tag: NN,
        label_index: 11,
        index: 133,
        word_index: 116,
        offset_begin: 425,
        offset_end: 432,
    },
    POSToken {
        word: "in",
        score: 0.9998982548713684,
        tag: IN,
        label_index: 6,
        index: 134,
        word_index: 117,
        offset_begin: 433,
        offset_end: 435,
    },
    POSToken {
        word: "your",
        score: 0.7870864868164063,
        tag: PRP,
        label_index: 17,
        index: 135,
        word_index: 118,
        offset_begin: 436,
        offset_end: 440,
    },
    POSToken {
        word: "NIF",
        score: 0.21826403493603763,
        tag: NN,
        label_index: 11,
        index: 136,
        word_index: 119,
        offset_begin: 441,
        offset_end: 444,
    },
    POSToken {
        word: "crate",
        score: 0.9713843464851379,
        tag: NN,
        label_index: 11,
        index: 138,
        word_index: 120,
        offset_begin: 445,
        offset_end: 450,
    },
    POSToken {
        word: "'",
        score: 0.9771125316619873,
        tag: POS,
        label_index: 16,
        index: 139,
        word_index: 121,
        offset_begin: 450,
        offset_end: 451,
    },
    POSToken {
        word: "s",
        score: 0.6749321222305298,
        tag: DT,
        label_index: 3,
        index: 140,
        word_index: 122,
        offset_begin: 451,
        offset_end: 452,
    },
    POSToken {
        word: "`",
        score: 0.6688801050186157,
        tag: NN,
        label_index: 11,
        index: 141,
        word_index: 123,
        offset_begin: 455,
        offset_end: 456,
    },
    POSToken {
        word: "Cargo",
        score: 0.5689875483512878,
        tag: NN,
        label_index: 11,
        index: 142,
        word_index: 124,
        offset_begin: 456,
        offset_end: 461,
    },
    POSToken {
        word: ".",
        score: 0.6731827259063721,
        tag: NNP,
        label_index: 12,
        index: 143,
        word_index: 125,
        offset_begin: 461,
        offset_end: 462,
    },
    POSToken {
        word: "toml",
        score: 0.19061735982491967,
        tag: NN,
        label_index: 11,
        index: 144,
        word_index: 126,
        offset_begin: 462,
        offset_end: 466,
    },
    POSToken {
        word: "`",
        score: 0.41062402725219727,
        tag: NNS,
        label_index: 14,
        index: 146,
        word_index: 127,
        offset_begin: 466,
        offset_end: 467,
    },
    POSToken {
        word: "on",
        score: 0.9963319301605225,
        tag: IN,
        label_index: 6,
        index: 147,
        word_index: 128,
        offset_begin: 468,
        offset_end: 470,
    },
    POSToken {
        word: "the",
        score: 0.9999924302101135,
        tag: DT,
        label_index: 3,
        index: 148,
        word_index: 129,
        offset_begin: 471,
        offset_end: 474,
    },
    POSToken {
        word: "`",
        score: 0.35348284244537354,
        tag: NN,
        label_index: 11,
        index: 149,
        word_index: 130,
        offset_begin: 475,
        offset_end: 476,
    },
    POSToken {
        word: "rustler",
        score: 0.20324191081416387,
        tag: NNP,
        label_index: 12,
        index: 150,
        word_index: 131,
        offset_begin: 476,
        offset_end: 483,
    },
    POSToken {
        word: "`",
        score: 0.5776707530021667,
        tag: NNS,
        label_index: 14,
        index: 152,
        word_index: 132,
        offset_begin: 483,
        offset_end: 484,
    },
    POSToken {
        word: "dependency",
        score: 0.9966633319854736,
        tag: NN,
        label_index: 11,
        index: 153,
        word_index: 133,
        offset_begin: 485,
        offset_end: 495,
    },
    POSToken {
        word: ".",
        score: NaN,
        tag: CC,
        label_index: 1,
        index: 154,
        word_index: 134,
        offset_begin: 495,
        offset_end: 496,
    },
]
